aws_auth.tf

```diff
@@ -66,21 +32,20 @@ data "template_file" "worker_role_arns" {
   }
 }
 
-data "template_file" "config_map_aws_auth" {
-  template = file("${path.module}/templates/config-map-aws-auth.yaml.tpl")
+resource "kubernetes_config_map" "aws_auth" {
+  count = var.manage_aws_auth ? 1 : 0
 
-  vars = {
-    worker_role_arn = join(
-      "",
-      distinct(
-        concat(
-          data.template_file.launch_template_worker_role_arns.*.rendered,
-          data.template_file.worker_role_arns.*.rendered,
-        ),
-      ),
-    )
-    map_users    = yamlencode(var.map_users),
-    map_roles    = yamlencode(var.map_roles),
-    map_accounts = yamlencode(var.map_accounts)
+  metadata {
+    name      = "aws-auth"
+    namespace = "kube-system"
+  }
+
+  data = {
+    mapRoles    = <<EOF
+${join("", distinct(concat(data.template_file.launch_template_worker_role_arns.*.rendered, data.template_file.worker_role_arns.*.rendered)))}
```

Comment:
Can we just remove these template files and use `yamlencode` juste like https://github.com/terraform-aws-modules/terraform-aws-eks/pull/355#issuecomment-507007419 ? This will prevent us to have configuration in multiple location and error in yaml template (wrong indentation by example).

Reply:
@barryib I think it is possible I just never could get it to work properly. Just tried it again. But by now the windows instances complicates stuff even more.